### PR TITLE
Feature/Enable change for propose attribute request item and creation of attribute

### PR DESCRIPTION
**Pull request by Siolto**

### CODE REVIEW

File: src/i18n/i18n_en.properties [hunk 1]
```diff
@@ -10,6 +10,11 @@
 #' = \u0027
 #" = \u0022
 
+
+popup.ok=OK
+popup.close=Close
```
Comment (jkoenig134) [inline src/i18n/i18n_en.properties:15]: missing translations

File: src/i18n/i18n_en.properties [hunk 2]
```diff
@@ -43,6 +48,7 @@ home.quickStart=Quick Start
 attribute.root.title=Root Attribute
 attribute.root.description=This root attribute is stored in your attribute repository and can be shared to various contacts.
 attribute.root.sharedWithDescription=This own attribute is shared with the following contacts:
+attribute.root.noshare=There are no eligible contacts to whom you can share this attribute.
```
Comment (jkoenig134) [inline src/i18n/i18n_en.properties:51]: missing translation

File: src/core/App.ts [hunk 3]
```diff
@@ -1,11 +1,7 @@
-import {
-    DatawalletSynchronizedEvent,
-    LocalAccountDTO,
-    MailReceivedEvent,
-    OnboardingChangeReceivedEvent
-} from "@nmshd/app-runtime"
+import { DatawalletSynchronizedEvent, MailReceivedEvent, OnboardingChangeReceivedEvent } from "@nmshd/app-runtime"
 import { DeviceDTO, RelationshipDVO } from "@nmshd/runtime"
 import { CoreId } from "@nmshd/transport"
+import { Dictionary } from "lodash"
```
Comment (jkoenig134) [inline src/core/App.ts:4]: I guess that import is unused?

**Review by jkoenig134 — APPROVED** [review]
LGTM

File: src/core/popups/CreateAttributePopup.controller.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    public async createAttribute() {
+        try {
+            const oModel = this.getModel()
+            const oOriginalName = oModel.getProperty("/name")
+
+            let valueType = this.valueTypeSelection.getSelectedKey()
+            const oValue = this.valueRenderer.getEditedValue()
+            if (oValue === "" || oValue === null || oValue === undefined) {
+                /*
```
Comment (vobu) [inline src/core/popups/CreateAttributePopup.controller.ts:65]: hmmm... if this -> do nothing, else -> do nothing? :)

File: src/core/popups/PopupController.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    private initialized: boolean = false
+    protected params: IAppPopupParams
+    protected view: View | undefined
+
+    public refreshPopup(params: IAppPopupParams, view: View) {
+        this.params = params
+        this.view = view
+        ;(this as any).oView = view
+        ;(this as any).oView.byId = (id: string) => {
```
Comment (vobu) [inline src/core/popups/PopupController.ts:25]: not understanding this: why redefine $view.byId? https://ui5.sap.com/#/api/sap.ui.core.mvc.View%23methods/byId

File: src/manifest.json [hunk 6]
```diff
@@ -4,7 +4,7 @@
         "id": "nmshd.app",
         "type": "application",
         "applicationVersion": {
-            "version": "2.3.9"
+            "version": "2.3.10"
```
Comment (vobu) [inline src/manifest.json:7]: please also check that all libs (such as `sap.ui.layout`) specified in `sap.ui5` -> `dependencies` -> `libs` are actually in use.  
and whether some that are in use are missing in the `dependencies`.  
-> big effect on build-time and bundle size...

**Review by vobu — COMMENTED** [review]
some things to look at, but not a whole lot

File: src/core/controls/requests/items/ProposeAttributeRequestItemRenderer.js [hunk 7]
```diff
@@ -128,32 +142,55 @@ sap.ui.define(
                 return responseParams
             },
 
+            // *****************************
+            // ***** private functions *****
+            // *****************************
+
+            _onAttributeChange(changedAttributePath) {
+                // results/0 is the already proposed item => we do nothing
+                if (changedAttributePath.includes("results/0")) {
```
[the remark this answers is not in the document]
Comment (Siolto) [reply src/core/controls/requests/items/ProposeAttributeRequestItemRenderer.js:151]: `this.selectedAttributePath` already has a default value of "results/0" so we do not have to set it here.

File: src/core/popups/PopupController.ts (hunk 5 above)
Comment (Siolto) [reply src/core/popups/PopupController.ts:25]: that's a tricky one and the naming might be a little bit of. `oView` is actually the Dialog. And to access controls inside of Fragments we would need to call `Fragment.byId(...)` all the time. We decided to stick to `byId` to not care if it is a view or fragment

File: src/manifest.json (hunk 6 above)
Comment (Siolto) [reply src/manifest.json:7]: removed A LOT of unused imports

File: src/core/controls/attributes/ValueRenderer.js [hunk 8]
```diff
@@ -101,7 +101,7 @@ sap.ui.define(
                     this.valueHints = object.results[0].valueHints
                 }
 
-                appLogger.log("Control invalidated through model " + oEvent.getSource().getId())
+                // appLogger.log("Control invalidated through model " + oEvent.getSource().getId())
```
[the remark this answers is not in the document]
Comment (Siolto) [reply src/core/controls/attributes/ValueRenderer.js:104]: at least I don't know about that. But that log statement should not be commented out :)

**Review by jkoenig134 — DISMISSED** [review]

File: src/core/controls/requests/items/ProposeAttributeRequestItemRenderer.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (vobu) [reply src/core/controls/requests/items/ProposeAttributeRequestItemRenderer.js:148]: anyway, nuisance, let's merge this so work can continue

**Review by vobu — APPROVED** [review]

**Pull request merged**